File: Queue/queue.c
#include <stdlib.h>
#include "queue.h"

void queueCreate(struct queueNode *q, int size){
    q->size = size;
    q->front = q->rear = 0;
    q->Q = (struct treeNode**)malloc(q->size * sizeof(struct treeNode*));
}

void enQueue(struct queueNode *q, struct treeNode *x){
    if((q->rear + 1) % q->size == q->front)
        puts("Queue is full!");
    else{
        q->rear = (q->rear + 1) % q->size;
        q->Q[q->rear] = x;
    }
}

struct treeNode* deQueue(struct queueNode *q){
    struct treeNode *x = NULL;
    if(q->front == q->rear)
        puts("Queue is empty!");
    else{
        q->front = (q->front + 1) % q->size;
        x = q->Q[q->front];
    }
    return x;
}

int isQueueEmpty(struct queueNode *q){
    return q->front == q->rear;
}

File: Java/Assignment_J/Queue.java
package Java.Assignment_J;

class QNode {
    int key;
    QNode next;

    QNode(int key) {
        this.key = key;
        this.next = null;
    }
}

public class Queue {
    QNode front;
    QNode rear;

    Queue(){
        this.front = null;
        this.rear = null;
    }

    void enqueue(int key){
        QNode node = new QNode(key);

        if(this.rear == null){
            this.front = node;
            this.rear = node;
            
            return;
        }

        this.rear.next = node;
        this.rear = node;
    }

    public void dequeue(){
        if(this.front == null){
            return;
        }

        QNode node = this.front;
        this.front = this.front.next;

        if(this.front == null){
            this.rear = null;
        }
    }
}
